source/rust_verify/src/context.rs

```diff
@@ -15,6 +17,7 @@ pub struct ErasureInfo {
 }
 
 type ErasureInfoRef = std::rc::Rc<std::cell::RefCell<ErasureInfo>>;
+type GlobalStrings = std::rc::Rc<std::cell::RefCell<HashMap<DefId, Arc<String>>>>;
```

Comment:
What is this? 
StrSlices by default are not lowered into air, at least not in any useful sense. 
When there is a "reveal" call, we then write the relevant air to represent that specific string. 
This holds all the String values along with their associated variable. So when VAR.reveal() is called, we look up the DefId for VAR and then retrieve the String, from here we can emit the air for this.

Reply:
migrated to vir/src/context.rs